employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/otp/OneTimePassword.java

```diff
@@ -0,0 +1,70 @@
+package org.optaweb.employeerostering.domain.otp;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Calendar;
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.constraints.Email;
+
+import org.optaweb.employeerostering.exception.OtpCreationException;
+
+@Entity
+public class OneTimePassword {
+    private static final Integer TOKEN_LENGTH = 6;
+    private static final Integer TOKEN_MAX = 10; // exclusive
+    private static final Integer EXPIRY_MINUTES = 5;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(unique = true)
+    private String email;
+
+    private String token;
+    private Integer retries = 3;
+
+    @Temporal(TemporalType.TIMESTAMP)
+    private Date expiry;
+
+    public OneTimePassword (@Email String email) throws OtpCreationException {
+        this.email = email;
+        this.token = generateToken();
+        this.expiry = calculateExpiry();
+    }
+
+    private static String generateToken() throws OtpCreationException {
+        try {
+            StringBuilder sb = new StringBuilder(TOKEN_LENGTH);
+            SecureRandom sr = SecureRandom.getInstance("SHA1PRNG");
```

Comment:
I'm not entirely sure this gives you the same `SecureRandom` instance, reading through the source. Given it doesn't, what you're basically getting is a non-seeded instance that will seed itself (but what with?) at the subsequent `nextInt()` call. Move this to class level field